app/src/main/java/com/example/gfood/screen/main/MainActivity.kt

```diff
@@ -14,4 +16,10 @@ class MainActivity : AppCompatActivity() {
         setContentView(R.layout.activity_main)
         addFragment(PageFragment.newInstance(), containerLayout.id)
     }
+
+    companion object {
+
+        fun newIntent(context: Context?): Intent =
+            Intent(context, MainActivity::class.java)
```

Comment:
sao lại truyền context vào đây e?
xem lại convention 2.10 - 2.11
```suggestion
    companion object {
        fun newIntent(context: Context?) = Intent(context, MainActivity::class.java)
```

Reply:
Dạ...Tại vì muốn intent 1 activity thì chúng ta phải truyền vào context của activity hiện tại và activity mới. Ở đây chúng ta tạo một hàm intent ở activity mới và gọi hàm này để start ở activity cũ. Nên chúng ta cần truyền vào context của activity cũ, nên ở đây chúng ta truyền vào context để activity cũ có thể start sang activity mới ạ